src/Service/PaymentService.php

```diff
@@ -102,6 +105,109 @@ public function checkSourceAuth(Source $source): bool
     }//end checkSourceAuth()
 
 
+    /**
+     * Get price from a single product.
+     *
+     * @param string $productId
+     *
+     * @return array       Product object.
+     */
+    private function getProductObject(string $productId): array
+    {
+        $productObject = $this->entityManager->getRepository('App:ObjectEntity')->find($productId);
+
+        return $productObject->toArray() ?? null;
+
+    }//end getProductObject()
+
+
+    /**
+     * Get price from a single product.
+     *
+     * @param array $product
+     *
+     * @return string|null Price.
+     */
+    private function getProductPrice(array $product)
+    {
+        if (isset($product['vertalingen'][0]['kosten'])) {
+            return $product['vertalingen'][0]['kosten'];
+        }//end if
+
+        return null;
+
+    }//end getProductPrice()
+
+
+    /**
+     * Get product prices from this marriage.
+     *
+     * @param array $huwelijk
+     *
+     * @return array $productPrices
+     */
+    public function getProductPrices(array $huwelijk): array
```

Comment:
Not so clean, maybe add todo: "split into functions, nesting level is higher than 3"?
Or just clean it up.